rebar.config.sample

```diff
@@ -97,21 +68,112 @@
 %% an application name, a version and the SCM details on how to fetch it (SCM
 %% type, location and revision).
 %% Rebar3 currently supports git and hg
-{deps, [app_name,
-        {rebar, "1.0.*"},
-        {rebar, ".*",
-         {git, "git://github.com/rebar/rebar.git"}},
-        {rebar, ".*",
-         {git, "git://github.com/rebar/rebar.git", "Rev"}},
-        {rebar, "1.0.*",
-         {git, "git://github.com/rebar/rebar.git", {branch, "master"}}},
-        {rebar, "1.0.0",
-         {git, "git://github.com/rebar/rebar.git", {tag, "1.0.0"}}},
-        {rebar, ".*",
-         {git, "git://github.com/rebar/rebar.git", {ref, "7f73b8d3650b41ffd53a199f3eda20985eda84e3"}}},
-        {app_name, ".*", {hg, "https://www.example.org/url"}}]}.
-
-%% == Pre/Post Command Hooks ==
+{deps, [
+    app_name,            % latest version of package
+    {rebar, "1.0.0"},    % version 1.0.0 of a package
+    {rebar,              % git, master branch of app, explicit
+     {git, "git://github.com/rebar/rebar.git", {branch, "master"}}},
+    {rebar,              % git, over HTTPS
+     {git, "https://github.com/rebar/rebar.git", {branch, "master"}}},
+    {rebar,              % git, tag
+     {git, "https://github.com/rebar/rebar.git", {tag, "1.0.0"}}},
+    {rebar,              % git, specific reference/hash
+     {git, "https://github.com/rebar/rebar.git", {ref, "7f73b8d6"}}},
+    {rebar,              % mercurial is also supported
+     {hg, "https://github.com/rebar/rebar.git", {tag, "1.0.0"}}},
+    %% Alternative formats, backwards compatible declarations
```

Comment:
I don't think we warn on the `raw` and version matches. I'll edit the unclear refs to show the warning  stuff more prominently.